### PR TITLE
Feature/add branch version to header file

### PR DESCRIPTION
**Pull request by JoergZeidler**
**blocks:** elbb/bb-buildingblock#24

### CODE REVIEW

File: gitversion/entrypoint.sh [hunk 1]
```diff
@@ -37,6 +37,17 @@ mkdir -p ${JSON_DIR}
...
+###### append branch version to json ######
+if [ "$(cat ${JSON_DIR}/gitversion.json | jq -r .BranchName)" == "master" ]; then
+    targetVersion="FullSemVer"
+else
+    targetVersion="InformationalVersion"
+fi
+targetVersionInfo=$(cat ${JSON_DIR}/gitversion.json | jq -r .${targetVersion})
+cat ${JSON_DIR}/gitversion.json | jq --arg value ${targetVersionInfo} '.+{BranchVersion: $value }' > ${JSON_DIR}/gitversion_append.json
+rm ${JSON_DIR}/gitversion.json
+mv ${JSON_DIR}/gitversion_append.json ${JSON_DIR}/gitversion.json
```
Comment (siredmar) [inline gitversion/entrypoint.sh]: the `rm` is not needed. the `mv` will overwrite the old file.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.6.2] - 2020.Q2
+
+### Added
+
+-   added branch name to gitversion json file
```
Comment (siredmar) [inline CHANGELOG.md]: 1. You are not adding the branch name, you are adding the `branch version`. See Readme for an explanation. Also you need to adapt the Readme that states the old `branch version` behavior. 
2. The change has a lot of more impact than just adding a value to the json file. The generated files (plain, env, and cpp) now also contain the `branch version`.

Comment (siredmar) [reply CHANGELOG.md]: Another thing not related to this PR:
Please remove the space between char and asterisk in this line https://github.com/elbb/bb-gitversion/blob/58073c46832e065c08141b7f39877d8c8edd2859/gitversion/templates/cpp/version.h.j2#L7
what i expect:
`constexpr const char* sometext`

**Review by siredmar — CHANGES_REQUESTED** [review]

File: gitversion/entrypoint.sh (hunk 1 above)
Comment (JoergZeidler) [reply gitversion/entrypoint.sh]: OK, will fix with next commit.

**Review by siredmar — APPROVED** [review]

**Pull request merged**